### ISSUE TITLE
Bug: iD changes back building=* to building=yes

### ISSUE THREAD

**Issue by maro-21** [openstreetmap/iD#7221]
When I add a POI to `building=retail`, iD changes back to `building=yes`.
When I add a hotel preset to `building=hotel`, iD changes back to `building=yes`.

**Comment by quincylvania** [comment]
I think this has come up before, but starting fresh, I do want to look into it further. It seems reasonable to keep the existing `building` tag if the target preset expects `building` but not a specific value.

**Comment by quincylvania** [comment]
Okay, so investigating this more, the issue is that `building` is sometimes a feature type and sometimes just an attribute.

When you change a line from River to Residential Road, you don't expect it to retain `waterway=river` as an attribute. So likewise it'd be unexpected to retain `building=house` when switching a House to a Seafood Shop. It's a bit tricky since `building=yes` is added as a default attribute, but that's fine since we can't assume the original building type would still be accurate.

In sum, I think the existing behavior is the safest and least confusing option for most iD mappers even if it occasionally leads to some dropped data. Here are some workarounds for avoiding this scenario:

- Map points-of-interest as points or landuse areas instead of combining them with buildings.
- Use the raw tag editor to add POI tags to buildings.

**Comment by scaidermern** [comment]
> So likewise it'd be unexpected to retain `building=house` when switching a House to a Seafood Shop.

No, that wouldn't be unexpected at all. The building value doesn't describe the current situation. It describes which feature this building was built for originally. So if you have a `building=church` and add a `shop=supermarket` the `building=church` has to stay, because the building remains being originally built as a church. That's why the current behaviour of iD is just wrong.

**Comment by maro-21** [comment]
scaidermern: I think it is ok, because like Quincy wrote, iD can't know about it. It doesn't read minds. For it you change one preset into another. The same behavior will be when you change from `building=construction` to `building=house`, `shop=vacant` to `shop=supermarket` and other.

I changed the way I edit with iD: if I add a preset with building, like hotel, I add it first and then change building=yes to building=hotel.
And I try to avoid combining buildings with POIs. After many years of mapping I realized this is a better way. Buildings don't change as often as POIs in these buildings and buildings can have many 3D tags. Also if one wants to add `start_date` tag, what will it be about - a building or a POI?
